src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -104,14 +104,17 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $opt
 				$integerRanges = TypeUtils::getIntegerRanges($offsetType);
 				if (count($integerRanges) > 0) {
 					foreach ($integerRanges as $integerRange) {
-						if ($integerRange->getMin() === null) {
+						$minRange = $integerRange->getMin();
+						if ($minRange === null || $minRange <= -self::ARRAY_COUNT_LIMIT) {
```

Comment:
I dont think the fix is correct. 

It reads like a range from a values smaller then -256 would no longer be possible?

Reply:
It wasn't possible before anyway, see next lines of code ... `if (count($scalarTypes) > 0 && count($scalarTypes) < self::ARRAY_COUNT_LIMIT) {...`

Reply:
I think its not the same thing to have a range of at max 256 elements vs only ranges between -256 and +256

Reply:
Other ideas? The problem is, we need to check this before the `range(` code, otherwise php will crash, Maybe we can avoid `range` and use a loop here?

Reply:
Hmm `max-min <= limit` could work?

Reply:
Yes, that was the idea. I already tried it, maybe you can take look at it, thanks.